### ISSUE TITLE
Creating a billing request without a payment fails

### ISSUE THREAD

**Issue by jkimbo** [gocardless/gocardless-pro-go#6]
Trying to create a billing request with only a mandate (and no payment) fails because the `payment_request` attribute is still being sent in the request. 

Creating a `BillingRequestCreateParams` with just a mandate request and customer converts to this JSON:
```json
{"billing_requests":{"links":{"customer":"CUSTOMER_ID"},"mandate_request":{"currency":"GBP"},"payment_request":{}}}
```

and the billing request endpoint returns a 422.

**Comment by vdandugc** [comment]
thanks for reporting the issue, will take a look at this.

**Comment by vdandugc** [comment]
hey @jkimbo thanks for that, I'm definitely interested in hearing out potential solutions, and yes i also thought the same about changing the type to pointer.  We have a template and an internal tool which generates the library from the template, the template would need to be tweaked in order to handle this.

**Comment by vdandugc** [comment]
an update on this, testing the fix, will update here once we push new version.

**Comment by vdandugc** [comment]
hi, apologies for the late update, this is now fixed in #4
